### PR TITLE
Initial async git prompt implementation

### PR DESCRIPTION
**Pull request by desyncr**
Code based off
http://www.anishathalye.com/2015/02/07/an-asynchronous-shell-prompt/

Fixes https://github.com/frmendes/geometry/issues/6

### TODO

  - [x] Use configuration variable for temporary file
  - [x] Add option to enable/disable async prompt
  - [x] Evaluate if there is a better option (avoid using a tmp file)
  - [x] Update README.md

### CODE REVIEW

**Comment by frm** [comment]
I'm assuming you'll add more to this PR or is the TODO supposed to be added as an ongoing process among different PRs?

**Comment by desyncr** [comment]
I'll add them to this PR and I created the TODO as to receive feedback on the items before start to work on them, if necessary.

**Comment by desyncr** [comment]
Done. Accepting feedback.

File: geometry.zsh [hunk 1]
```diff
@@ -48,13 +48,17 @@ GEOMETRY_PROMPT=$(prompt_geometry_colorize $GEOMETRY_COLOR_PROMPT $GEOMETRY_SYMB
...
+PROMPT_GEOMETRY_GIT_ASYNC=${PROMPT_GEOMETRY_GIT_ASYNC:-true}
 PROMPT_GEOMETRY_EXEC_TIME=${PROMPT_GEOMETRY_EXEC_TIME:-false}
 PROMPT_GEOMETRY_COLORIZE_SYMBOL=${PROMPT_GEOMETRY_COLORIZE_SYMBOL:-false}
 PROMPT_GEOMETRY_COLORIZE_ROOT=${PROMPT_GEOMETRY_COLORIZE_ROOT:-false}
 PROMPT_VIRTUALENV_ENABLED=${PROMPT_VIRTUALENV_ENABLED:-false}
 PROMPT_GEOMETRY_COMMAND_MAX_EXEC_TIME=${PROMPT_GEOMETRY_COMMAND_MAX_EXEC_TIME:-5}
 PROMPT_GEOMETRY_GIT_TIME_SHORT_FORMAT=${PROMPT_GEOMETRY_GIT_TIME_SHORT_FORMAT:-true}
 
+# Misc configurations
+GEOMETRY_ASYNC_PROMPT_TMP_FILENAME=${GEOMETRY_ASYNC_PROMPT_TMP_FILENAME:-/tmp/geometry-promp-git-info-}
```
Comment (frm) [inline geometry.zsh]: Can you fix the typo? `promp` should be `prompt`. Also, why the final `-`?

File: geometry.zsh [hunk 2]
```diff
@@ -270,6 +278,35 @@ prompt_geometry_clear_timestamp() {
   unset prompt_geometry_command_exec_time
 }
 
+GEOMETRY_ASYNC_PROMPT_PROC=0
+prompt_geometry_setup_async_prompt() {
+    function precmd() {
```
Comment (frm) [inline geometry.zsh]: `precmd` runs before each prompt, but does it interfere with other `precmd` hook functions?

**Review by frm — CHANGES_REQUESTED** [review]
Looks pretty good and clean! Tested a bit and it's much better.

Just the typo thing and that one question. I can't merge since there are conflicts as well.

File: geometry.zsh (hunk 1 above)
Comment (desyncr) [reply geometry.zsh]: Sure thing. The trailing dash is just for file name formatting: `/tmp/geometry-prompt-git-info-1234` where `1234` is the process id.

File: geometry.zsh (hunk 2 above)
Comment (desyncr) [reply geometry.zsh]: It doesn't look like it interferes at all. Nevertheless I'm gonna change to use preferred format `add-zsh-hook`.

**Comment by desyncr** [comment]
Done. Fixed typo and replaced `precmd` function for `add-zsh-hook`.

Squashed and fixed merge issue.

**Review by frm — APPROVED** [review]

**Pull request merged**